add 관련 메소드 추가

lns13301 opened the pull request:
- Node 객체를 생성할 inner Class 생성
- add() 추가 // index, element 둘 다
- addAll() 추가 // index, element 둘 다
- 기타 size(), isEmpty 추가
- getNode 추가

lns13301 committed:
add 관련 메소드 추가

- Node 객체를 생성할 inner Class 생성
- add() 추가 // index, element 둘 다
- addAll() 추가 // index, element 둘 다
- 기타 size(), isEmpty 추가
- getNode 추가

LTTTTTE commented on src/main/java/MyLinkedList.java:
```diff
@@ -0,0 +1,221 @@
+    private Node next;
+    private Node tail;
+    private Node head;
+
```
생성자가 안보임, 필드변수 초기화시키거나 생성자에서 해줄 필요

lns13301 replied:
생성자 생성 완료

LTTTTTE commented on src/main/java/MyLinkedList.java:
```diff
@@ -0,0 +1,221 @@
+            this.data = data;
+            this.next = null;
+        }
+        public Node(E data, Node next){
```
Node의 두번째 생성자는 무엇을 뜻하는 지 모르겟음

lns13301 replied:
두 번째 생성자의 경우 add(index) 메소드 호출 시 이전 노드를 불러와 그 노드에 새로운 노드를 한 번에 연결해주기위해서 사용함

LTTTTTE reviewed (commented):
아직 Add 부분은 못보겠음

lns13301 committed:
코드 오류 수정

- 테스트 코드의 생성확인 메소드가 isNotNull()이 아닌 크기가 0인지로 비교하고 있어서 발생한 오류 수정
- 코드 생성자 추가

lns13301 committed:
코드 오류 수정

- tail값을 반대로 입력한 것 같아서 수정

LTTTTTE commented:
Node 에도 Generic 표기 필요

LTTTTTE reviewed (commented):
MyLinkedList 내에 필드변수중 Node next는 무슨애인지

lns13301 committed:
Generic 추가 및 set메소드 추가

- Node 클래스의 Generic을 추가
- set 메소드 구현
- Node next에 대한 답변
   - next에는 다음에오는 노드가 담긴다. 이를 통해서 head부터 tail까지 연결고리가 만들어지는 것이다.

lns13301 committed:
Generic 추가

- 코드 안에 Generic을 추가해주지 않아서 추가함

lns13301 committed:
branch오류수정중

lns13301 commented:
LinkedList는 새로운 Repository 생성 후 작업을 진행하도록 하겠음
